docs/platform/tutorials/image-processing/other-services.md

```diff
@@ -0,0 +1,195 @@
+    ]
+  },
+  "TagValues": {}
+}
+```
+
+In this example there are 2 cars, and 1 truck giving a `vehicles` count of 3.
+
+The main code is:
+
+``` python
+def on_dataframe_received_handler(stream_consumer: qx.StreamConsumer, df: pd.DataFrame):
+    # List of vehicle columns
+    vehicle_columns = ['car', 'bus', 'truck', 'motorbike']
+
+    # Calculate the total vehicle count based on existing columns
+    total_vehicle_count = df.apply(lambda row: sum(row.get(column, 0) for column in vehicle_columns), axis=1)
+
+    # Store vehicle count in the data frame
+    df["vehicles"] = total_vehicle_count
+    stream_producer = topic_producer.get_or_create_stream(stream_id = stream_consumer.stream_id)
+    # Publish data frame to the producer stream
+    stream_producer.timeseries.buffer.publish(df)
+```
+
+You can find out more about pandas DataFrames in the [pandas documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html){target=_blank}.
+
+## Max vehicles
+
+The max vehicles service takes the total vehicle count and finds the maximum value over a one day window. This value is made available to the Data API service. The message passed to the Data API has the following format:
+
+``` json
+{
+  "Epoch": 0,
+  "Timestamps": [
+    1694088514402644000
+  ],
+  "NumericValues": {
+    "max_vehicles": [
+      8
+    ]
+  },
+  "StringValues": {},
+  "BinaryValues": {},
+  "TagValues": {
+    "window_start": [
+      "2023-09-06 12:08:12.394372"
+    ],
+    "window_end": [
+      "2023-09-07 12:08:12.394372"
+    ],
+    "window": [
+      "1d 0h 0m"
+    ],
+    "cam": [
+      "JamCams_00001.08959"
+    ]
+  }
+}
+```
+
+You can see the exact time window is recorded, along with the maximum vehicle count during that time window. This provides a crude measure of the capacity of the road. This capacity can then be used by the UI to calculate a percentage of capacity. For example, if there are 8 cars on a road, and the maximum seen is 10, then the road is considered to be at 80% capacity, and this is displayed on the UI, as shown in the following screenshot:
+
+![Road capacity](../image-processing/images/road-capacity.png)
+
+This service uses [state](https://quix.io/docs/client-library/state-management.html), as you need to save the maximum count reached during the time window. 
+
+## Data API
+
+The Data API provides these endpoints:
+
+* max_vehicles
+* detected_objects
+
+These are used by the UI to obtain and display the data on the web interface.
+
+### Max Vehicles 
+
+Returns the maximum number of "vehicles" seen on a camera, where vehicles is one of cars, buses, trucks, or motorbikes.
+
+For a `GET` on the endpoint `/max_vehicles`, the response is a dictionary item per camera:
+
+* Key=camera name
+* Value=max vehicle count
+
+Example response JSON:
+
+``` json
+{
+    "JamCams_00001.01251":2.0,
+    "JamCams_00001.01252":1.0
+}
+```
+
+This service is implemented as a simple Flask web app hosted in Quix.
```

Comment:
Another small thing, but is this correct terminology? Might be, but I know Flask is a server, rather than a web app. I would tend to say something like "it's an API running behind a Flask server".

Reply:
Yes, I think it's OK. Flask docs call it an application. I called it a web app because it has the endpoints of a typical web app. For "web server" I think of things like Nginx. I've linkified the "Flask web app" text to the Flask Quickstart, so they can see exactly what I mean though.